pkg/proxy/nftables/proxier.go

```diff
@@ -1422,6 +1533,55 @@ func (proxier *Proxier) syncProxyRules() {
 			}
 		}
 
+		if svcInfo.SessionAffinityType() == v1.ServiceAffinityClientIP {
+			// Generate the per-endpoint affinity sets
+			for _, ep := range allLocallyReachableEndpoints {
+				epInfo, ok := ep.(*endpointInfo)
+				if !ok {
+					proxier.logger.Error(nil, "Failed to cast endpointsInfo", "endpointsInfo", ep)
+					continue
+				}
+				activeAffinitySets.Insert(epInfo.affinitySetName)
+			}
+		}
+
+		// All the following operations are service-chain related and may be skipped if no svc or endpoint
+		// changes are required.
+		if svcTx == nil {
+			continue
+		}
+
+		if usesFWChain {
```

Comment:
To keep the overall structure similar to iptables where possible, please put this after the `if usesExternalTrafficChain` section